### PR TITLE
chore: improve setup docs

### PR DESCRIPTION
**Pull request by justinmoon**

### CODE REVIEW

File: docs/dev-env.md [hunk 1]
```diff
@@ -12,7 +12,12 @@ and follow the instructions instead of reading this document.
 
 ## Clone repository
 
-Clone this repository locally, with `git clone <repo-url>`, then `cd <repo-dir>`,
+Clone the fedimint git repository locally and cd into it:
+
+```
+git clone git@github.com:fedimint/fedimint.git
+cd fedimint
+```
```
Comment (justinmoon) [inline docs/dev-env.md:20]: Easier to copy-paste. If user wants to clone over https they can figure make the change themselves.

File: docs/dev-env.md [hunk 2]
```diff
@@ -33,10 +38,11 @@ it ensures consistent and reproducible environment for all developers.
 
 ### Install Nix
 
-If you don't have it set up already,
-follow the instructions at: https://nixos.org/download.html
+You have 2 options to install nix:
+* [The official installer](https://nixos.org/download.html)
+* The [Determinate Nix Installer](https://zero-to-nix.com/start/install) which is maintained by a 3rd party, but is a little more user-friendly.
```
Comment (justinmoon) [inline docs/dev-env.md:43]: People have trouble installing nix. Give them a 2nd option which worked well for me.

File: docs/dev-running.md [hunk 3]
```diff
@@ -1,25 +1,8 @@
-## Running Fedimint for dev testing
-
-Fedimint consists of three kinds of executables:
-
-* **Federation nodes** - servers who form the mint by running a consensus protocol
-* **Lightning gateways** - allows users send and receive over Lightning by bridging between the mint and an LN node
-* **User clients** - handles user communication with the mint and the gateway
-
```
Comment (justinmoon) [inline docs/dev-running.md:8]: This is out-of-date and not really necessary to describe initially. Let's get them to a working tmuxinator environment and then they can explore the services themselves.

File: docs/dev-running.md [hunk 4]
```diff
@@ -1,25 +1,8 @@
...
-- The [Rust toolchain](https://www.rust-lang.org/tools/install) to build and run the executables
-- The [Nix package manager](https://nixos.org/download.html) for managing build and test dependencies
-
-Clone and `cd` into the Fedimint repo:
-
-```shell
-git clone git@github.com:fedimint/fedimint.git
-cd fedimint
-```
+Fedimint's developer environment is build around the [Nix package manager](https://nixos.org). Once setup, you can execute one `nix develop` command to create a completely deterministic development, test and build environment with all required dependencies installed and pinned to exact versions. Detailed setup instructions are available in [dev-env.md](./dev-env.md).
```
Comment (justinmoon) [inline docs/dev-running.md:3]: Old instructions kinda suggested that they should install rust outside nix. Let's be more clear about forcing them to install nix, but also explain why it's worth it.

File: docs/dev-env.md (hunk 2 above)
Comment (Maan2003) [reply docs/dev-env.md:43]: > ### Enable nix flakes

This is not needed for Determinate nix installer.

Comment (justinmoon) [reply docs/dev-env.md:43]: Thanks. Updated that section to mention this.

**Review by Maan2003 — DISMISSED** [review]

**Review by dpc — APPROVED** [review]
Nice.

**Review by dpc — APPROVED** [review]

**Comment by GetPsyched** [comment]
> If the Nix installation is in multi-user mode, don’t forget to restart the nix-daemon.

It may be helpful to also suggest the command to restart the nix-daemon below this line. A user might have to go to an external webpage to figure out how to restart daemons if its their first time doing so.

Something along the lines of:
> If the Nix installation is in multi-user mode, don’t forget to restart the nix-daemon; you can do so by the following command:
> ```
> sudo systemctl restart nix-daemon.service
> ```

**Review by okjodom — APPROVED** [review]

**Pull request merged**